src/main/java/com/farao_community/farao/gridcapa/job_launcher/JobLauncherService.java

```diff
@@ -20,24 +26,59 @@
  */
 @Service
 public class JobLauncherService {
+    private static final Logger LOGGER = LoggerFactory.getLogger(JobLauncherService.class);
     private static final String RUN_BINDING = "run-task";
 
+    private final JobLauncherConfigurationProperties jobLauncherConfigurationProperties;
+    private final RestTemplateBuilder restTemplateBuilder;
     private final StreamBridge streamBridge;
     private final Logger jobLauncherEventsLogger;
 
-    public JobLauncherService(StreamBridge streamBridge, Logger jobLauncherEventsLogger) {
+    public JobLauncherService(JobLauncherConfigurationProperties jobLauncherConfigurationProperties,
+                              RestTemplateBuilder restTemplateBuilder, StreamBridge streamBridge,
+                              Logger jobLauncherEventsLogger) {
+        this.jobLauncherConfigurationProperties = jobLauncherConfigurationProperties;
+        this.restTemplateBuilder = restTemplateBuilder;
         this.streamBridge = streamBridge;
         this.jobLauncherEventsLogger = jobLauncherEventsLogger;
     }
 
-    public void launchJob(TaskDto taskDto) {
-        if (taskDto.getStatus() == TaskStatus.READY
-            || taskDto.getStatus() == TaskStatus.SUCCESS
-            || taskDto.getStatus() == TaskStatus.ERROR) {
-            jobLauncherEventsLogger.info("Task launched on TS {}", taskDto.getTimestamp());
-            streamBridge.send(RUN_BINDING, Objects.requireNonNull(taskDto));
-        } else {
-            jobLauncherEventsLogger.warn("Failed to launch task with timestamp {} because it is not ready yet", taskDto.getTimestamp());
+    public boolean launchJob(String timestamp) {
+        LOGGER.info("Received order to launch task {}", timestamp);
+        String requestUrl = getUrlToRetrieveTaskDto(timestamp);
+        LOGGER.info("Requesting URL: {}", requestUrl);
+        ResponseEntity<TaskDto> responseEntity = restTemplateBuilder.build().getForEntity(requestUrl, TaskDto.class);
+        TaskDto taskDto = responseEntity.getBody();
+        if (taskDto != null) {
+            String taskId = taskDto.getId().toString();
+            // propagate in logs MDC the task id as an extra field to be able to match microservices logs with calculation tasks.
+            // This should be done only once, as soon as the information to add in mdc is available.
+            MDC.put("gridcapa-task-id", taskId);
+
+            if (responseEntity.getStatusCode() == HttpStatus.OK) {
+                restTemplateBuilder.build().put(getUrlToUpdateTaskStatusToPending(timestamp), TaskDto.class);
```

Comment:
it would be more readable if put just before the stream bridge.send(run...), because it makes confusion between the status you just put , and the status you are comparing with to check if the run is possible

Reply:
True ! Even the case where the computation will not be launched (in the else) it does not work, because if we change the status to pending and then don't do any action it will stay stuck in pending status.